frontend/src/components/ColumnChartWidget.tsx

```diff
@@ -41,6 +45,20 @@ const ColumnChartWidget = (props: Props) => {
   const previewWidth = 480;
   const fullWidth = 960;
 
+  const columnChartRef = useRef(null);
+  useEffect(() => {
+    if (columnChartRef && columnChartRef.current) {
+      const yAxisMap = (columnChartRef.current as CategoricalChartWrapper).state
+        .yAxisMap;
+      if (yAxisMap && yAxisMap.length) {
+        setYAxisMargin(
+          yAxisMap[0].niceTicks.toLocaleString().match(/,/g)?.length +
+            pixelsByCharacter
+        );
+      }
+    }
+  }, [columnChartRef, columnChartRef.current, chartLoaded]);
+
```

Comment:
hard to follow what is going on here. Is there any way we can make this easier to read? It also seems to be repeated in the LineChartWidget. Maybe extracting it into a hook would help?

Reply:
Added explanation, not extracting it to a hook here.